enterprise/ha/src/test/java/org/neo4j/kernel/ha/transaction/TransactionPropagatorTest.java

```diff
@@ -85,4 +96,68 @@ public SlavePriority getReplicationStrategy()
         // THEN
         verify( logger, times( 1 ) ).warn( anyString() );
     }
+
+    @Test
+    public void shouldPrioritizeAscendingIfAsked() throws Exception
+    {
+        // GIVEN
+        Configuration propagator = TransactionPropagator.from( new Config( stringMap( tx_push_strategy.name(), fixed_ascending.name() )));
+        SlavePriority strategy = propagator.getReplicationStrategy();
+
+        // WHEN
+        Iterable<Slave> prioritize = strategy.prioritize( asList( slave( 1 ), slave( 0 ), slave( 2 ) ) );
+
+        // THEN
+        assertThat( Iterables.toList( prioritize ), equalTo( asList( slave( 0 ), slave( 1 ), slave( 2 ) ) ) );
+    }
+
+    @Test
+    public void shouldPrioritizeDescendingIfAsked() throws Exception
+    {
+        // GIVEN
+        Configuration propagator = TransactionPropagator.from( new Config( stringMap( tx_push_strategy.name(), fixed.name() )));
+        SlavePriority strategy = propagator.getReplicationStrategy();
+
+        // WHEN
+        Iterable<Slave> prioritize = strategy.prioritize( asList( slave( 1 ), slave( 0 ), slave( 2 ) ) );
+
+        // THEN
+        assertThat( Iterables.toList(prioritize), equalTo( asList( slave( 2 ), slave( 1 ), slave( 0 ) ) ) );
+    }
+
+    private Slave slave( final int id )
+    {
+        return new Slave()
```

Comment:
It's makes the test failures much easier to read though, since that class has a nice toString that includes which id it has, and it means `slave(1).equals(slave(1))`, which is not true if I use a mock, making the assertions easier.